app/models/manageiq/providers/foreman/provider.rb

```diff
@@ -20,7 +20,7 @@ class ManageIQ::Providers::Foreman::Provider < ::Provider
 
   delegate :api_cached?, :ensure_api_cached, :to => :connect
 
-  before_create :ensure_managers
+  before_validation :ensure_managers
```

Comment:
I think this should be `before_save`, right?  Otherwise we could wind up updating the name on the managers before running the validations on the name.